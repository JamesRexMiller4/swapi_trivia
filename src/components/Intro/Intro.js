import React from 'react';
import './Intro.scss';

const Intro = (props) => {
  let mov = props.movie();
  return (
    <section className='text-section'>
      <h1 className='title'>{mov.title}</h1>
      <p className='text'>{mov.text}</p>
    </section>
  )
}

export default Intro;
