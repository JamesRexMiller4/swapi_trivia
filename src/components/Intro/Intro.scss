html {
  background: {
    image: url('../../images/stars2.jpg');
    size: auto;
  }
  height: 100vh;
  margin: 0;
  width: 100%;
}
