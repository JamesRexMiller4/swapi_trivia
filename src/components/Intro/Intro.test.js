import React from 'react';
import ReactDOM from 'react-dom';
import Intro from './Intro';
import { shallow } from 'enzyme';

describe('Intro', () => {
  let wrapper, movie;

  beforeEach(() => {
    movie = jest.fn(() => {
      return {title: 'some title',
      text: 'some text'}
    })
    wrapper = shallow(<Intro movie={movie}/>)

  });

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Intro movie={movie}/>, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('should match the snapshot with all the data passed through', () => {
    expect(wrapper).toMatchSnapshot()
  });
  
});
