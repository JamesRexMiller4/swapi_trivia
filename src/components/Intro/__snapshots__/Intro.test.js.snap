// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Intro should match the snapshot with all the data passed through 1`] = `
<section
  className="text-section"
>
  <h1
    className="title"
  >
    some title
  </h1>
  <p
    className="text"
  >
    some text
  </p>
</section>
`;
